src/components/Toggle/Toggle.jsx

```diff
@@ -0,0 +1,93 @@
+import React from "react";
+import PropTypes from "prop-types";
+import classNames from "classnames";
+import NOOP from "lodash/noop";
+import { useToggleState } from "@react-stately/toggle";
+import { useSwitch } from "@react-aria/switch";
+import { VisuallyHidden } from "@react-aria/visually-hidden";
+import { BASE_TOGGLE_CLASS_NAME } from "./ToggleConstants";
+import ToggleText from "./ToggleText";
+import "./Toggle.scss";
+import { useToggle } from "../../hooks/useToggle";
+
+const Toggle = ({
+  id,
+  componentClassName,
+  isDefaultSelected,
+  isSelected,
+  onChange,
+  value,
+  name,
+  isDisabled,
+  ariaLabel,
+  ariaControls,
+  isHideLabels
+}) => {
+  const { inputProps } = useToggle({
```

Comment:
not sure I understood what triggers the toggle,
I see it uses checked, but when there is some callback like

onCheckedChange or whatever supposed to be here